src/flamegpu/util/detail/JitifyCache.cu

```diff
@@ -81,74 +82,138 @@ std::string loadFile(const path &filepath) {
     ifs.close();
     return rtn;
 }
-}  // namespace
 
-std::mutex JitifyCache::instance_mutex;
-std::unique_ptr<KernelInstantiation> JitifyCache::compileKernel(const std::string &func_name, const std::vector<std::string> &template_args, const std::string &kernel_src, const std::string &dynamic_header) {
-    NVTX_RANGE("JitifyCache::compileKernel");
-    // Init runtime compilation constants
-    static std::string env_inc_fgp2 = std::getenv("FLAMEGPU2_INC_DIR") ? std::getenv("FLAMEGPU2_INC_DIR") : "";
-    static bool header_version_confirmed = false;
-    static std::string env_cuda_path = std::getenv("CUDA_PATH") ? std::getenv("CUDA_PATH") : "";
-    if (env_inc_fgp2.empty()) {
+/**
+ * Find the cuda include directory.
+ * Throws exceptions if it can not be found.
+ * @return the path to the CUDA include directory.
+ */
+std::string getCUDAIncludeDir() {
+    // Define an array of environment variables to check in order
+    std::array<const std::string, 2> ENV_VARS { "CUDA_PATH", "CUDA_HOME" };
```

Comment:
This feels like something vs gives warnings about (don't think it likes const std::string in collections for some reason).

Reply:
CI didn't seem to trigger so it looks like this isnt' quite what MSVC gets angry about, but I do seem to recall something along these lines being problematic in the past.